homeassistant/components/binary_sensor/lupusec.py

```diff
@@ -42,4 +45,8 @@ def is_on(self):
     @property
     def device_class(self):
         """Return the class of the binary sensor."""
-        return self._device.generic_type
+        if self._device.generic_type in DEVICE_CLASSES:
+            return self._device.generic_type
+        else:
+            _LOGGER.error('Binary sensor device class not known')
```

Comment:
We don't want side effects in entity state properties.

Reply:
Explicitly `return None` in this case. We want to be consistent when returning values.

Reply:
So I add a line under the 'LOGGER' with 'return None'?

Reply:
No the logging should be removed. We don't allow side effects in entity state properties. If you copy the example I posted, that should be ok.

Reply:
Return None if device class not found. Remove else and outdent the last return.

https://en.m.wikipedia.org/wiki/Side_effect_(computer_science)